### PR TITLE
Add blame functionality

### PR DESCRIPTION
**Pull request by ballessay**
See issue #9

### CODE REVIEW

File: include/git2cpp/blame.h [hunk 1]
```diff
@@ -1,23 +1,25 @@
-﻿#pragma once
+#pragma once
 
-#include <memory>
+#include <git2/blame.h>
```
Comment (AndreyG) [inline include/git2cpp/blame.h]: Let's leave forward declarations instead of `#include`.

File: include/git2cpp/blame.h [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
             : blame_(blame)
         {}
 
-        git_blame_hunk const * get_hunk_byline(size_t lineno) const;
-
     private:
         struct Destroy { void operator() (git_blame *) const; };
-
-        std::unique_ptr<git_blame, Destroy> blame_;        
+        std::unique_ptr<git_blame, Destroy> blame_;
```
Comment (AndreyG) [inline include/git2cpp/blame.h]: This change is redundant.

File: src/blame.cpp [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+        return git_blame_get_hunk_byindex(blame_.get(), index);
+    }
+
+    const git_blame_hunk* Blame::hunk_byline(size_t lineno) const
     {
         return git_blame_get_hunk_byline(blame_.get(), lineno);
     }
 
-    void Blame::Destroy::operator()(git_blame* blame) const
+    void Blame::Destroy::operator() (git_blame * blame) const
```
Comment (AndreyG) [inline src/blame.cpp]: This change is redundant.

File: include/git2cpp/blame.h [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
 
 namespace git
 {
     struct Blame
     {
+        uint32_t hunk_count() const;
+
+        const git_blame_hunk * hunk_byindex(uint32_t index) const;
+
+        const git_blame_hunk * hunk_byline(size_t lineno) const;
```
Comment (AndreyG) [inline include/git2cpp/blame.h]: I'm OK without the prefix `get_`, but examples/blame.cpp should be fixed in this case.

File: include/git2cpp/blame.h (hunk 1 above)
Comment (ballessay) [reply include/git2cpp/blame.h]: You're right.

File: include/git2cpp/blame.h (hunk 2 above)
Comment (ballessay) [reply include/git2cpp/blame.h]: This just removes the empty line, which is consistent with diff.h

File: src/blame.cpp (hunk 3 above)
Comment (ballessay) [reply src/blame.cpp]: The * alignment is consistent with other function definitions

File: include/git2cpp/blame.h (hunk 4 above)
Comment (ballessay) [reply include/git2cpp/blame.h]: I'll fix the example.

**Pull request merged**